src/main/java/net/neoforged/neoforge/registries/datamaps/DataMapType.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.registries.datamaps;
+
+import com.google.common.base.Preconditions;
+import com.mojang.serialization.Codec;
+import java.util.Objects;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceKey;
+import net.minecraft.resources.ResourceLocation;
+import net.neoforged.neoforge.common.conditions.ConditionalOps;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A registry data map contains data-driven object that can be attached to a registry object. <p>
+ * Data maps are registered to the {@link RegisterDataMapTypesEvent}. <p>
+ * They are loaded from JSON files located at:
+ *
+ * <pre>
+ * <code>:mapNamespace/data_maps/:registryNamespace/:registryPath/:mapPath.json</code>
```

Comment:
Maybe `<map namespace>` instead? `:` can be confusing in the context of resource locations.

(You can use `<pre>{@code ... }</pre>` to escape things correctly.)